### PR TITLE
Files [OSF-7451]

### PR DESCRIPTION
**Pull request by alexschiller**
Add files routes

[#OSF-7451]
https://openscience.atlassian.net/browse/OSF-7451

### CODE REVIEW

File: swagger-spec/files/version_detail.yaml [hunk 1]
```diff
@@ -1 +1,50 @@
...
+  tags:
+    - Files
+  operationId: files_version_detail
+  responses:
+    '200':
+      description: OK
+      schema:
+        type: array
+        items:
+          $ref: 'version_definition.yaml'
```
Comment (caseyrollins) [inline swagger-spec/files/version_detail.yaml:40]: Where's `version_definition.yaml`?

Comment (alexschiller) [reply swagger-spec/files/version_detail.yaml:40]: That's why it is a WIP pr :). Was multitasking and missed it. Added.

File: swagger-spec/files/definition.yaml [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+        description: 'Whether or not the current user is allowed to post comments.'
+      guid:
+        type: string
+        readOnly: true
+        description: 'Global unique identifier (GUID) for this file (if one has been assigned).'
+      checkout:
+        type: string
+        readOnly: true
+        description: 'SOON TO BE DEPRECATED, see relationships checkout.'
+      tags:
```
Comment (caseyrollins) [inline swagger-spec/files/definition.yaml:76]: The tags property can actually be:
```
tags:
  type: array
  items: 
    type: string
  description: ''
```

File: swagger-spec/files/definition.yaml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        readOnly: true
+        description: 'An array of tags on the file (strings).'
+      size:
+        type: integer
+        readOnly: true
+        description: 'Size of the file in bytes.'
+  relationships:
+    type: object
+    readOnly: true
+    description: ''
```
Comment (caseyrollins) [inline swagger-spec/files/definition.yaml]: 'URLs to other entities or entity collections that have a relationship to the file entity.'

File: swagger-spec/files/versions_list.yaml [hunk 4]
```diff
@@ -1 +1,60 @@
 # /files/{file_id}/versions/
+get:
+  summary: Retrieve file versions
```
Comment (caseyrollins) [inline swagger-spec/files/versions_list.yaml]: "List all file versions"

I've been inconsistent about this (sorry), but I think I've fixed it everywhere now. List views should use "List..." and detail views should use "Retrieve..."

File: swagger-spec/institutions/definition.yaml [hunk 5]
```diff
@@ -8,7 +8,7 @@ properties:
   type:
     type: string  
     readOnly: true
-    description: The type identifier of the institution identity (`institutions`).
+    description: The type identifier of the institution entity (`institutions`).
```
Comment (caseyrollins) [inline swagger-spec/institutions/definition.yaml]: 👍

**Review by caseyrollins — CHANGES_REQUESTED** [review]
This looks great! A few extremely minor changes and this will be RTM. 🦄

**Review by caseyrollins — APPROVED** [review]

**Pull request merged**